### ISSUE TITLE
Bug: `no-missing-link-fragments` fails if heading contained emoji

### ISSUE THREAD

**Issue by adamlui** [eslint/markdown#439]
### Environment

ESLint version: v9.29.0
@eslint/markdown version: 6.6.0
Node version: v22.15.0
npm version: v11.3.0
Operating System: win32 10.0.19045

### Which language are you using?

commonmark

### What did you do?

<details>
<summary>Configuration</summary>

```js
import js from '@eslint/js'
import importPlugin from 'eslint-plugin-import'
import markdown from '@eslint/markdown'
import stylisticJS from '@stylistic/eslint-plugin'
import yml from 'eslint-plugin-yml'

export default [
    { ignores: ['**/package-lock.json'] },
    {
        files: ['**/*.{js,mjs}'],
        languageOptions: { ecmaVersion: 'latest', sourceType: 'script' },
        plugins: { 'import': importPlugin, 'js-styles': stylisticJS },
        rules: {
            ...js.configs.recommended.rules,
            ...importPlugin.flatConfigs.recommended.rules,
            'indent': 'off', 'no-unexpected-multiline': 'off', 'key-spacing': 'off', // allow whitespace anywhere
            'js-styles/no-trailing-spaces': 'error', // ...except at ends of lines
            'js-styles/max-len': ['error', { 'code': 120, // limit lines to 120 chars except if containing...
                'ignoreComments': true, 'ignoreStrings': true, // ...trailing/own-line comments, quoted strings...
                'ignoreTemplateLiterals': true, 'ignoreRegExpLiterals': true }], // ...or template/regex literals
            'js-styles/no-extra-semi': 'error', // disallow unnecessary semicolons
            'quotes': ['error', 'single', // enforce single quotes...
                { 'allowTemplateLiterals': true }], // ...except backticks to avoid escaping quotes
            'comma-dangle': ['error', 'never'], // enforce no trailing commas in arrays or objects
            'no-constant-condition': 'off', // allow constant conditions
            'no-empty': 'off', // allow empty blocks
            'no-inner-declarations': 'off', // allow function declarations anywhere
            'no-useless-escape': 'off', // allow all escape chars cause ESLint sucks at detecting truly useless ones
            'no-unused-vars': ['error', { 'caughtErrors': 'none' }] // allow unused named args in catch blocks
        }
    },
    {
        files: ['**/*.md'], language: 'markdown/commonmark', plugins: { markdown },
        rules: {
            ...markdown.configs.recommended[0].rules,
            'markdown/heading-increment': 'off', // allow headings to skip levels
            'markdown/fenced-code-language': 'off', // allow code blocks w/ no language specified
            'markdown/no-missing-label-refs': 'off', // allow missing label references
            'markdown/no-multiple-h1': 'off', // allow multi H1s
            'markdown/require-alt-text': 'off' // allow missing img alts
        }
    },
    { files: ['**/*.mjs'], languageOptions: { sourceType: 'module' }},
    { files: ['**/*.{yaml,yml}'], ...yml.configs['flat/standard'][1] }
]
```
</details>

```md
...
# Contents

- [Compatibility](#compatibility)
- [Scripts](#scripts)
  - [Ad blockers](#-ad-blockers)
  - [Amazon](#-amazon)
  - [Brave](#-brave)
  - [ChatGPT](#-chatgpt)
  - [Discord](#-discord)
  - [Display](#-display)
  - [DuckDuckGo](#-duckduckgo)
  - [GitHub](#-github)
  - [Google](#-google)
  - [Just Eat (UK)](#-just-eat-uk)
  - [Links](#-links)
  - [Media](#-media)
  - [Navigation](#-navigation)
  - [Passwords](#-passwords)
  - [Privacy](#-privacy)
  - [Text-to-speech](#-text-to-speech)
  - [Translation](#-translation)
  - [YouTube](#-youtube)
- [Tutorials](#tutorials)
- [Community](#community)
- [Additional Catalogues](#additional-catalogues)
- [Contributing](#contributing)
- [License](#license)
...
```

### What did you expect to happen?

No error

### What actually happened?

![Image](https://github.com/user-attachments/assets/11ed019d-9f44-49aa-8840-b2b053f79d8b)

### Link to Minimal Reproducible Example

https://github.com/awesome-scripts/awesome-userscripts/actions/runs/15822360880/job/44594316899

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

When including emojis in headings GitHub strips them from the fragments which are still valid and working
